### PR TITLE
Sex estimation module

### PR DESCRIPTION
**Pull request by 15E601**
This branch implements the sex-estimation module. Requires a change in the RDFBones app to appear in the sidebar

### CODE REVIEW

**Comment by cuboideum** [comment]
The Phaleron-SexEstimation ontology extension has been provided as a dependency in the Phaleron App Ontology repository with commit https://github.com/AnthroGraph/Phaleron-AppOntology/commit/daf70e23cb5669273bbb3645479d29cf57e09868.

**Comment by zarquon42b** [comment]
I just loaded it with the latest Ontology. The module is there but after creating an instance no triples have been generated and therefore there is no associated template page. @15E601 can you please check that.

What I did: 

1. Set RDFBones-App to sex-est branch 
2. Set PBP-App to sex-estimation branch
3. build and load latest ontology provided by @cuboideum (see above)
4. Go to Sex Estimation module and select a SI and click on "Save"
5. You get directed to an error page (see below)

 
Selection_178

**Comment by zarquon42b** [comment]
Info: The WHERE pattern fails at this line:
`?myQueriedMorphology rdfs:subClassOf ?myRS1 .`
This indicates that the resulting class is missing the necessary statement. I also do not find those restrictions in the latest version of the PBP-App ontology. Maybe something went wrong with the ontology merge?

File: data/templates/http%3A%2F%2Fwww.metaphacts.com%2Fresource%2FPBP_SexEst_CreateSexEst.html [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+  ?myTarget owl:intersectionOf ?what .
+  
+  ?what rdf:first ?myQueriedMorphology .
+  ?what rdf:rest ?rest .
+  ?rest rdf:first ?desc .
+  ?desc a owl:Restriction . 
+  ?desc owl:onProperty ?myQueriedProperty ;
+        owl:onClass ?ROI .
+ 
+ ?myQueriedMorphology rdfs:subClassOf ?myRS1 .
```
Comment (zarquon42b) [inline data/templates/http%3A%2F%2Fwww.metaphacts.com%2Fresource%2FPBP_SexEst_CreateSexEst.html:61]: This line fails with the current version of the App ontologie

**Comment by zarquon42b** [comment]
... the above pattern is included [here](https://github.com/AnthroGraph/PBP-App/blob/3912bfc53df90cff59bc2483d2cea1be23dc13f5/data/templates/http%253A%252F%252Fwww.metaphacts.com%252Fresource%252FPBP_SexEst_CreateSexEst.html#L61)

**Comment by 15E601** [comment]
I have confirmed the bug. The issue is with the ROBOT script: it only merges the phaleron-se ontology. Further required are the Klales, Phenice, Walker, and Standards sex estimation ontologies, as these contain the value specification restrictions the query is looking for. Uploading these ontologies along with the ROBOT's phaleron-app ontology file fixed the issue in my testing.
The ROBOT script should be altered to include these ontologies. As I understand the script, this would mean adding the missing ontologies as dependencies, as they are hierarchically above the phaleron-se ontology.

**Comment by cuboideum** [comment]
The missing dependencies have been added to the app ontology with commit [bfd8680744ad6461207862384fbdb9938895d5ed](https://github.com/AnthroGraph/Phaleron-AppOntology/commit/bfd8680744ad6461207862384fbdb9938895d5ed).

**Pull request merged**